chatminer/chatparsers.py

```diff
@@ -6,8 +6,7 @@
 from pathlib import Path
 logging.basicConfig(level=logging.INFO)
 
-
-class WhatsAppParser():
+class Parser():
```

Comment:
The methods `_parse_message`, `_read_file_into_list`, and `parse_file_into_df` are present in both derived classes. The code would be even cleaner if you introduced these methods as abstract methods (see [here](https://www.geeksforgeeks.org/abstract-classes-in-python/)) in the parent class to make the logic more rigid.

Reply:
true, but `_parse_message` and `_read_file_into_list` are not part of `Parser`'s public interface. In theory, there could be parsers (e.g. for telegram messages) that do not require these methods. I'd therefore advocate for only making `parse_file_into_df` an abstract method of `Parser`. However, it is your project, so if you really want me to, I can also include `_parse_message` and `_read_file_into_list` as abstract methods.